### PR TITLE
chore: Automated releases based off conventional commits

### PR DESCRIPTION
**Pull request by EAlainMG**
Commitlint + husky to locally enforce conventional commits, release-please used to automate releases accordingly.

On a Mac, the .husky/ directory requires the executable flag to be manually set. I've already done this and pushed the changes, and ensured a newly cloned repo functions expectedly after initialising the dependencies.

### CODE REVIEW

**Comment by EAlainMG** [comment]
Will update token variable in release.yml workflow shortly, as being told I was given an incorrect one

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        contents: write
+        pull-requests: write
+    steps:
+      - name: Release Please  
+        uses: google-github-actions/release-please-action@v3
+        with:
+          token: ${{ secrets.RA_GH_RELEASE_TOKEN  }}
+          release-type: simple
+          package-name: actions
+          changelog-types: '[{"type":"feat","section":"Features","hidden":false},{"type":"fix","section":"Bug Fixes","hidden":false},{"type":"chore","section":"Miscellaneous","hidden":false},{"type":"refactor","section":"Refactor","hidden":false}]'
```
Comment (o-leolleo) [inline .github/workflows/release.yml:24]: Hey just a small note to add a new line here

Comment (o-leolleo) [reply .github/workflows/release.yml:24]: and on the other places this red sign shows up

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Changelog
```
Comment (o-leolleo) [inline CHANGELOG.md:1]: Hey out of curiosity how was this changelog generated?

Comment (EAlainMG) [reply CHANGELOG.md:1]: @o-leolleo Change log was automatically generated, but I used this task/opportunity to play around with github revision control so the changelog probably consists of history I rewrote.. apologies I didn't consider this

File: .github/workflows/release.yml (hunk 1 above)
Comment (EAlainMG) [reply .github/workflows/release.yml:24]: Will do !

Comment (EAlainMG) [reply .github/workflows/release.yml]: @o-leolleo all newline errors now resolved

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  release-please:
+    if: "${{ !contains(github.event.pull_request.title || github.event.issue.title, 'chore(main): release') }}"
```
Comment (EAlainMG) [inline .github/workflows/release.yml:12]: @o-leolleo 
release.yml workflow updated to ignore self-created PRs. 
Seems that's it for this repo regarding unit tests for now, unless I'm missing something.. as I'm not seeing any tests setup. 
Let me know, thanks

Comment (EAlainMG) [reply .github/workflows/release.yml:12]: resolved in priv mgs

**Review by o-leolleo — APPROVED** [review]

**Review by o-leolleo — CHANGES_REQUESTED** [review]
Hey @EAlainMG I believe we can add the steps outline [here](https://github.com/google-github-actions/release-please-action?tab=readme-ov-file#creating-majorminor-tags) for also having v1, v1.x and v1.x.x style of tags

**Pull request closed without merging**